source/common/router/router.cc

```diff
@@ -769,18 +769,25 @@ Filter::createConnPool(Upstream::ThreadLocalCluster& thread_local_cluster) {
     factory = &config_.router_context_.genericConnPoolFactory();
   }
 
-  bool should_tcp_proxy = false;
-
+  using UpstreamProtocol = Envoy::Router::GenericConnPoolFactory::UpstreamProtocol;
+  UpstreamProtocol upstream_protocol = UpstreamProtocol::HTTP;
   if (route_entry_->connectConfig().has_value()) {
     auto method = downstream_headers_->getMethodValue();
-    should_tcp_proxy = (method == Http::Headers::get().MethodValues.Connect);
-
+    if (method == Http::Headers::get().MethodValues.Connect) {
+      upstream_protocol = UpstreamProtocol::TCP;
+    }
+    if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.enable_connect_udp_support") &&
```

Comment:
Can / should this `if` and the next be `else if`? I *think* that the three branches are mutually exclusive, but I'm not sure.

Reply:
The existing if statements were quite complicated to me as well so I refactored them in the new commit. Thanks for pointing this out.